gemd/demo/cake.py

```diff
@@ -916,7 +1063,21 @@ def find_name(name, material):
     run_key = md5.hexdigest()
 
     # Crawl tree and annotate with uids; only add ids if there's nothing there
-    recursive_foreach(cake, lambda obj: obj.uids or obj.add_uid(DEMO_SCOPE, obj.name + run_key))
+    name_count = dict()
+
+    def _disambig(name):
+        nonlocal name_count
```

Comment:
This non-local pattern looks to work, but I think the slightly more pythonic way would be to add `name_count` to the argument list of `_disambig`.

Reply:
Migrated to using a closure & generator.